configuration/configmgr/configmgrlib/mod_template_support/EnvModTemplate.cpp

```diff
@@ -220,124 +282,195 @@ void EnvModTemplate::parseOperation(const rapidjson::Value &operation)
 
     if (action == "create_node")
     {
-        pOp = parseCreateNodeOperation(operation);
+        pOp = std::make_shared<OperationCreateNode>();
     }
-    else if (action == "noop" || action == "modify_node")
+    else if (action == "modify_node")
     {
-        pOp = parseOperationNoop(operation);
+        pOp = std::make_shared<OperationModifyNode>();
+    }
+    else if (action == "find_node")
+    {
+        pOp = std::make_shared<OperationFindNode>();
+    }
+    else if (action == "delete_node")
+    {
+        pOp = std::make_shared<OperationDeleteNode>();
     }
     else
     {
         throw TemplateException("Unsupported operation '" + action + "' found", false);
     }
 
+    //
+    // Get the parent based on presence of key (schema validates presence of one of these)
+    auto it = operation.FindMember("parent_path");
+    if (it != operation.MemberEnd())
+    {
+        pOp->m_path = trim(it->value.GetString());
+    }
+    else
+    {
+        it = operation.FindMember("parent_nodeid");
+        pOp->m_parentNodeId = trim(it->value.GetString());
+    }
+
+    //
+    // Parse the data section
+    auto dataIt = operation.FindMember("data");
+    if (dataIt != operation.MemberEnd())
+    {
+        parseOperationCommonData(dataIt->value, pOp);
+
+        //
+        // Parse specific operation type values
+        if (action == "create_node")
+        {
+            std::shared_ptr<OperationCreateNode> pCreateOp = std::dynamic_pointer_cast<OperationCreateNode>(pOp);
+            pCreateOp->m_nodeType = dataIt->value.FindMember("node_type")->value.GetString();
+        }
+        else if (action == "find_node")
+        {
+            it = dataIt->value.FindMember("create_if_not_found");
+            if (it != dataIt->value.MemberEnd())
+            {
+                std::shared_ptr<OperationFindNode> pFindOp = std::dynamic_pointer_cast<OperationFindNode>(pOp);
+                pFindOp->m_createIfNotFound = it->value.GetBool();
+            }
+        }
+    }
+
     m_operations.emplace_back(pOp);
 }
 
 
-void EnvModTemplate::parseOperationCommon(const rapidjson::Value &operation, std::shared_ptr<Operation> pOp)
+void EnvModTemplate::parseOperationCommonData(const rapidjson::Value &operationData, std::shared_ptr<Operation> pOp)
 {
     rapidjson::Value::ConstMemberIterator it;
+    auto dataObj = operationData.GetObject();
+
 
     //
-    // Either a node ID or a path
-    it = operation.FindMember("parent_nodeid");
-    if (it != operation.MemberEnd())
-        pOp->setParentNodeId(it->value.GetString());
-    else
-        pOp->setPath(operation.FindMember("path")->value.GetString());
+    // Get the count (optional, default is 1)
+    it = dataObj.FindMember("count");
+    if (it != dataObj.MemberEnd())
+        pOp->m_count = trim(it->value.GetString());
```

Comment:
No because they could be substitution variables in the form {{varname}}. For that reason, they are kept in their raw for until used whereby a substitution and conversion to a number is done. At that time an exception, during execution of the template, can be thrown if there is a problem.